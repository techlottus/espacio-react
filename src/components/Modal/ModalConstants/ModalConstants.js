import React, { useEffect, useRef, useState } from "react";
import { Button } from "../../Button/Button";
import { constantsModalBtn } from "../../../constants/Procedures.constant";
import "./ModalConstants.scss";
import { CardTypeConstant } from "../../Card/CardTypeConstant";
import { useSelector } from "react-redux";
import { downloadFile } from "../../../helpers/download";
import { notiObs } from "../../../observables/notificationObs";
import { typesNoti } from "../../../types/typeNoti";
import env from "../../../enviroment/environment";
import { sendInfoTM } from "../../../tagging/services/sendInfoTagManager";
import {
  closeRecordTypeStudyRecordTag,
  downloadRecordTypeStudyRecordTag,
} from "../../../tagging/flows/procedures/studyRecordTag";

export const ModalConstants = ({ isShow, onClose, HandleDemo, outside }) => {
  const { modal } = useSelector((state) => state.procedureStudyRecord);
  const { texts } = useSelector((state) => state.texts);

  const modalRef = useRef();

  const [isModal, setIsModal] = useState(false); 

  useEffect(() => {
    setIsModal(isShow);
  }, [isShow]);

  const [desktop, setDesktop] = useState(window.innerWidth > 991);
  const updateMedia = () => {
    setDesktop(window.innerWidth > 991);
  };

  useEffect(() => {
    window.addEventListener("resize", updateMedia);
    return () => window.removeEventListener("resize", updateMedia);
  });
  useEffect(() => {
    if (isShow) {
      document.body.style.overflowY = "hidden";
    } else {
      document.body.style.overflowY = "auto";
    }
  }, [isShow]);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (
        modalRef.current &&
        !modalRef.current.contains(event.target) &&
        window.innerWidth > 991
      ) {
        outside();
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const handleModal = () => {
    return (
      <div className="modal-constants">
        <div className="modalcontent" ref={modalRef}>
          <div className="modalheader">
            <div className="title">
              {texts?.procedures?.modalConstants?.title}
            </div>
            <span
              className="material-icons icon"
              onClick={() => {
                onClose();
                sendInfoTM(
                  window,
                  closeRecordTypeStudyRecordTag,
                  "modal_t_close"
                );
              }}
            >
              close
            </span>
          </div>
          <div className="btnconstants">
            <div className="btnprimary">
              <Button
                data={{
                  ...constantsModalBtn,
                  isExpand: window.innerWidth < 991,
                }}
                onClick={() => {
                  downloadFile(
                    "tipos-constancias.pdf",
                    `${env.contentAssets}docs/constancia/utc/tipos-constancias.pdf`
                  )
                    .then(() => {
                      notiObs.next({
                        type: typesNoti.success,
                        text: texts.procedures.downloadPDFSuccess,
                      });
                      sendInfoTM(
                        window,
                        downloadRecordTypeStudyRecordTag,
                        "modal_t_demo"
                      );
                    })
                    .catch(() => {
                      notiObs.next({
                        type: typesNoti.error,
                        text: texts.procedures.downloadPDFError,
                      });
                    });
                  onClose();
                }}
              />
            </div>
          </div>
          <div className="modalbody">
            {modal.map((dataModal, i) => {
              const data = {
                title: dataModal.label,
                text: dataModal.description,
                image: desktop ? dataModal.imageUrl : null,
              };
              return <CardTypeConstant data={data} key={i} />;
            })}
          </div>
        </div>
      </div>
    );
  };

  return <>{isModal ? handleModal() : ""}</>;
};
