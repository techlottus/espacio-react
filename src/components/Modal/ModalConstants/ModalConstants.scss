@import '../../../App.scss';
@import "../Modal.scss";

.modal-constants {
  @extend .modal-container;
  align-items: inherit;
  .modalcontent {
    height: 80vh;
    margin-top: 5vh;
    overflow-y: auto;
    @media only screen and (min-width: map-get($grid-breakpoints, s)) {
      width: calc(60% - #{pixeltorem(48px)});
      height: 80vh;
    }
    .modalheader {
      display: flex;
      justify-content: space-between;
    }
    .modalbtns {
      margin-bottom: pixeltorem(16px);
    }
  }
}
