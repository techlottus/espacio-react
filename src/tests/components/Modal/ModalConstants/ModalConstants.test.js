import React from "react";
import { mount } from "enzyme";
import { ModalConstants } from "../../../../components/Modal/ModalConstants/ModalConstants";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { BrowserRouter } from "react-router-dom/cjs/react-router-dom.min";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };

let initState = {
  texts: {},
  procedureStudyRecord: {
    modal: {},
  },
};

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <BrowserRouter>
        <ModalConstants history={historyMock} />
      </BrowserRouter>
    </Provider>
  );
};

describe("Test <ModalConstants />", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });
  describe("set isModal", () => {
    test("set isModal to true", () => {
      const wrapper = handleMount(initState);
    });
  });
});
