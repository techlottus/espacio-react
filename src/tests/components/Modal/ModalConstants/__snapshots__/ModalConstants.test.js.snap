// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ModalConstants /> should init component 1`] = `""`;
